code/modules/projectiles/boxes_magazines/internal/revolver.dm

```diff
@@ -7,6 +7,16 @@
 /obj/item/ammo_box/magazine/internal/cylinder/rev38/rubber
 	ammo_type = /obj/item/ammo_casing/c38/match/bouncy
 
+/obj/item/ammo_box/magazine/internal/cylinder/rev38/random
+	start_empty = TRUE //We have to handle adding dynamic ammo types on init
+
+/obj/item/ammo_box/magazine/internal/cylinder/rev38/random/Initialize(mapload)
+	var/obj/item/ammo_casing/c38/boolet
+	for(var/i in 1 to max_ammo)
+		boolet = pick(typesof(/obj/item/ammo_casing/c38) -/obj/item/ammo_casing/c38/dart -/obj/item/ammo_casing/caseless/mime)
```

Comment:
You should store the result of `typesof(/obj/item/ammo_casing/c38) -/obj/item/ammo_casing/c38/dart -/obj/item/ammo_casing/caseless/mime)` into a variable instead of trying `pick(that_one)` every iteration.

Reply:
That completely changes the logic and will result in a gin with 6 of the same bullet.

Reply:
it doesn't. I am not saying the result of pick. I am saying assigning thr list manipilation result.

Reply:
Oh, yes I can do that.

Reply:
Done and re-tested

![image](https://github.com/user-attachments/assets/06cb6b18-15c9-42e7-8447-a7c232141e4a)